src/utils.rs

```diff
@@ -1,149 +1,90 @@
+use num_enum::TryFromPrimitive;
 
-/// Function to determine gender based on the unique number (BBB)
-pub fn determine_gender(unique_number: u32) -> String {
-    if (0..=499).contains(&unique_number) {
-        "Male".to_string()
-    } else {
-        "Female".to_string()
-    }
+#[derive(Debug, PartialEq, TryFromPrimitive)]
+#[repr(u32)] // Ensures each variant is represented by a u32
+pub enum Gender {
+    Male = 0,
+    Female = 1,
 }
 
-pub fn map_region(region_code: u32) -> &'static str {
-    match region_code {
-        1 => "Stranci u BiH",
-        2 => "Stranci u Crnoj Gori",
-        3 => "Stranci u Hrvatskoj",
-        4 => "Stranci u Severnoj Makedoniji",
-        5 => "Stranci u Sloveniji",
-        7 => "Beograd",
-        8 => "Vojvodina",
-        9 => "Kosovo",
-        10 => "Bihać, Bosna i Hercegovina",
-        11 => "Doboj, Bosna i Hercegovina",
-        12 => "Goražde, Bosna i Hercegovina",
-        13 => "Livno, Bosna i Hercegovina",
-        14 => "Mostar, Bosna i Hercegovina",
-        15 => "Prijedor, Bosna i Hercegovina",
-        16 => "Sarajevo, Bosna i Hercegovina",
-        17 => "Tuzla, Bosna i Hercegovina",
-        18 => "Zenica, Bosna i Hercegovina",
-        20 => "Podgorica, Crna Gora",
-        21 => "Bar, Crna Gora",
-        22 => "Budva, Crna Gora",
-        23 => "Herceg Novi, Crna Gora",
-        24 => "Cetinje, Crna Gora",
-        25 => "Nikšić, Crna Gora",
-        26 => "Berane, Crna Gora",
-        27 => "Bijelo Polje, Crna Gora",
-        28 => "Pljevlja, Crna Gora",
-        30 => "Osijek, Hrvatska",
-        31 => "Varaždin, Hrvatska",
-        32 => "Zagreb, Hrvatska",
-        33 => "Karlovac, Hrvatska",
-        34 => "Lika, Hrvatska",
-        35 => "Sisak, Hrvatska",
-        36 => "Dalmacija, Hrvatska",
-        40 => "Bitolj, Severna Makedonija",
-        41 => "Kumanovo, Severna Makedonija",
-        42 => "Ohrid, Severna Makedonija",
-        43 => "Prilep, Severna Makedonija",
-        44 => "Skoplje, Severna Makedonija",
-        45 => "Strumica, Severna Makedonija",
-        46 => "Tetovo, Severna Makedonija",
-        47 => "Veles, Severna Makedonija",
-        48 => "Štip, Severna Makedonija",
-        50 => "Slovenija",
-        70 => "Beograd",
-        71 => "Beograd",
-        72 => "Kragujevac",
-        73 => "Niš",
-        74 => "Leskovac",
-        75 => "Zaječar",
-        76 => "Smederevo",
-        77 => "Mačva",
-        78 => "Čačak",
-        79 => "Užice",
-        80 => "Novi Sad",
-        81 => "Sombor",
-        82 => "Subotica",
-        83 => "Vrbas",
-        84 => "Kikinda",
-        85 => "Zrenjanin",
-        86 => "Pančevo",
-        87 => "Vršac",
-        88 => "Sremska Mitrovica",
-        90 => "Priština, Kosovo",
-        91 => "Kosovska Mitrovica, Kosovo",
-        92 => "Peć, Kosovo",
-        93 => "Đakovica, Kosovo",
-        94 => "Prizren, Kosovo",
-        95 => "Kosovska Kamenica, Kosovo",
-        _ => "Nepoznat region",
+impl Gender {
```

Comment:
When converting from one type to another in idiomatic Rust it is better to implement From or TryFrom traits, in this case TryFrom since u32 can be greater then 999 and it can fail.
```rust
impl TryFrom<u32> for Gender {
    fn try_from(item: u32) -> Self;
}```